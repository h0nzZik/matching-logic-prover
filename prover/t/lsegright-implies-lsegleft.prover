claim \implies( \and( listSegmentRight( variable("H") { ArrayIntInt }
                                      , variable("X") { Int }
                                      , variable("Y") { Int }
                                      , variable("F") { SetInt }
                                      )
                     )
              , \and( listSegmentLeft(variable("H") { ArrayIntInt }
                                     , variable("X") { Int }
                                     , variable("Y") { Int }
                                     , variable("F") { SetInt }
                                     )
                     )
              )

strategy kt ; normalize ; ( ( simplify ; right-unfold-Nth(0, 0) ; smt )
                          | ( kt-solve-implications(smt)
                            ; ( ( normalize ; kt ; ( ( right-unfold-Nth(0, 1)
                                                     ; right-unfold-Nth(0, 0) 
                                                     ; normalize
                                                     ; instantiate-existentials
                                                     ; smt
                                                     )
                                                   | ( right-unfold-Nth(0, 1) 
                                                     ; instantiate-existentials
                                                     ; instantiate-aux
                                                     ; kt-solve-implications
                                                          ( instantiate-existentials
                                                          ; simplify ; smt
                                                          )
                                                     ; normalize
                                                     ; smt
                                                     )
                                                   | wait
                                                   )
                                ) | wait
                              )
                            )
                          | wait
                          )

// strategy search-bound(5)
/*
strategy kt ; ( ( right-unfold-Nth(0, 0) ; simplify ; direct-proof )
              | ( direct-proof )
              | ( kt ; ( ( simplify ; right-unfold-Nth(0, 1) ; right-unfold-Nth(0, 0) ; substitute-equals-for-equals ; direct-proof)
                       | ( simplify ; direct-proof)
                       | right-unfold-Nth(0, 1) ; simplify ; direct-proof
                       | noop
                       )
                )
              | ( noop )
              )
*/
