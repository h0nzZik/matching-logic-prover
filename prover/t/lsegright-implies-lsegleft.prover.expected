<prover>
  <goal>
    <id>
      0
    </id>
    <active>
      true
    </active>
    <parent>
      -1
    </parent>
    <k>
      \implies ( \and ( listSegmentRight ( variable ( "H" ) { ArrayIntInt } , variable ( "X" ) { Int } , variable ( "Y" ) { Int } , variable ( "F" ) { Set } , .Patterns ) , .Patterns ) , \and ( listSegmentLeft ( variable ( "H" ) { ArrayIntInt } , variable ( "X" ) { Int } , variable ( "Y" ) { Int } , variable ( "F" ) { Set } , .Patterns ) , .Patterns ) )
    </k>
    <strategy>
      success
    </strategy>
    <trace>
      .
    </trace>
  </goal>
</prover>
